animated_sprite.cpp

```diff
@@ -0,0 +1,113 @@
+#include "animated_sprite.h"
+#include <iostream>
+
+// AnimatedSprite()
+AnimatedSprite::AnimatedSprite(std::string sprite_sheet_name,
+                               SDL_Rect first_sprite_coord_location,
+                               SDL_Renderer* renderer,
+                               float time_to_update_sprite,
+                               SDL_Rect destination)
+    : Sprite(sprite_sheet_name, first_sprite_coord_location, renderer),
+      time_to_update(time_to_update_sprite),
+      current_animation_once(false),
+      current_animation(""),
+      frame_index_(0),
+      visible_(true),
+      destination_(destination) {
+  this->SetupAnimations();
+}
+
+// TimeToUpdateFrame()
+void AnimatedSprite::TimeToUpdateFrame() {
+  if (elapsed_time_ >= time_to_update) {
+    if (frame_index_ < (int)animations_[this->current_animation].size() - 1) {
+      frame_index_++;
```

Comment:
You just want to loop from 0 to animations.size. You can do this with the mod operator:
`frame_index = frame_index % animations_[current_animation].size();`